### PR TITLE
Add pipeline to produce animated gage timeseries in R

### PR DESCRIPTION
**Pull request by cnell-usgs**
Description
-----------
Adding subdirectory to recreate the gage timeseries gif ([from here](https://labs.waterdata.usgs.gov/visualizations/gages-through-the-ages/index.html#/)) in R with updated data. The original animation is done with svg and javascript, so this was recreated by adapting the [`gage_age_gif`](https://github.com/usgs-makerspace/gages-through-the-ages/tree/main/gage_age_gif) pipeline to match it. 

The original `gage_age_gif` was made using `scipiper`, so  a new `targets` pipeline is added for this gif separately rather than build on the old one. Some of the mapping steps use a shift and rotate process using `sp`. Edits were make to both `gage_age_gif` and `gage_timeseries_gif` to maintain this process. It'd be better if it were adapted to `sf`, but was sufficient for this purpose. It's not necessary to review the `scipiper` pipeline at this time.

To recreate:
- Add the `national-flow-observations` output of `active_flow_gages_summary_wy.rds` in`gage_timeseries_gif/data/`. 
- From `gage_timeseries_gif` run `tar_make(gif_final)`

This requires `gifciscle` to be installed (https://www.lcdf.org/gifsicle/). To skip the gifsicle step, set `reduce = FALSE` on [line 91 of `_targets.R`](https://github.com/usgs-makerspace/gages-through-the-ages/blob/f046eb35ac724fc02e8739004c269871b0d7f403/gage_timeseries_gif/_targets.R#L91).

Expected output:
gage-timeseries-optimized

### CODE REVIEW

File: gage_timeseries_gif/src/plot_utils.R [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+  # sites active in a given year
+  gages_active <- sites.sf |>
+    filter(site_no %in% yr_gages$site)
+  
+  # plot map
+  gage_map <- gages_active |>
+    ggplot() +
+    geom_sf(data = states.sf, fill = '#ededed',
+            linewidth = 0.25, color = 'white') +
+    #geom_sf(color = 'white', size = 0.2) +
```
Comment (aaarcher-usgs) [inline gage_timeseries_gif/src/plot_utils.R:89]: Clean up code throughout by deleting commands that are commented out.

**Review by aaarcher-usgs — CHANGES_REQUESTED** [review]
This took 8 minutes to run locally, and created the gif as expected. I don't have gifsicle on my computer, so had to run with `reduce = FALSE`. 

I have some suggestions/questions here and there about the code. I think there are a couple places where some redundant code can be removed, and I noticed a bit of commented-out code throughout that can be removed for cleaning up the pipeline.

File: gage_timeseries_gif/src/map_utils.R [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  map <- maps::map(..., fill = TRUE, plot = FALSE)
+  IDs <- sapply(strsplit(map$names, ":"), function(x) x[1])
+  map.sp <- map2SpatialPolygons(map, IDs=IDs, proj4string=CRS("+proj=longlat +datum=WGS84"))
+  map.sp.t <- spTransform(map.sp, CRSobj = proj.string)
+  return(map.sp.t)
+}
+
+#' @param locations a data.frame with dec_long_va and dec_lat_va
+points_sp <- function(locations){
+  library(dplyr)
```
[the remark this answers is not in the document]
Comment (cnell-usgs) [reply gage_timeseries_gif/src/map_utils.R:19]: yeah totally. didn't catch that when this script was copied over from `gage_age_gif`

File: gage_timeseries_gif/src/map_utils.R [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  
+  shift_details <- list(...)
+  
+  gages_info <- readRDS(gage_data) %>%
+    pull(site) %>% 
+    dataRetrieval::readNWISsite() %>% 
+    group_by(site_no) %>% 
+    # parse huc_cd to 2 digits, and rename to huc to stay consistent
+    # 3/19/2020 - discovered that some sites have more than one HUC code, not sure how that
+    # is possible, but filtering to the last one (most recent) for now
```
[the remark this answers is not in the document]
Comment (cnell-usgs) [reply gage_timeseries_gif/src/map_utils.R:59]: I don't follow this. That's an old comment that probably needs to be cleaned up

[the remark this answers is not in the document]
Comment (cnell-usgs) [reply gage_timeseries_gif/src/map_utils.R:59]: I'm not totally sure how to interpret it. @lindsayplatt might have some insight on the comment

File: gage_timeseries_gif/src/plot_utils.R [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+#' @param yr year being shown
+plot_gage_timeseries <- function(gage_melt, yr){
+  
+  font_fam <-'Source Sans Pro'
+  sysfonts::font_add_google(font_fam, regular.wt = 300, bold.wt = 700)
+  showtext::showtext_opts(dpi = 300)
+  showtext::showtext_auto(enable = TRUE)
+  
+  # store vars for plotting
+  active_year <- as.numeric(as.character(yr))
```
[the remark this answers is not in the document]
Comment (cnell-usgs) [reply gage_timeseries_gif/src/plot_utils.R:9]: lol

**Comment by cnell-usgs** [comment]
Suggested edits have been addressed.

**Review by aaarcher-usgs — APPROVED** [review]
Looks good!

File: gage_timeseries_gif/src/map_utils.R (hunk 3 above)
[the remark this answers is not in the document]
Comment (lindsayplatt) [reply gage_timeseries_gif/src/map_utils.R:55]: Oooo fun edge case - literally!

Those two HUCs are actually right next to each other (see HUC info [here](https://water.usgs.gov/GIS/regions.html)). The big HUC02s represent California (`18`) and the "Great Basin" (`16`), which share a border. If you drill down and look at just those two HUC08s and where the site is, it makes sense that people changed the HUCs depending on the year. Looks like it is right on the border, so either could be true!

<details>
  <summary>Code to make the plot</summary>

```r
library(sf)
library(ggplot2)

huc_2014 <- nhdplusTools::get_huc(id = "18020129", type="huc08")
huc_2016 <- nhdplusTools::get_huc(id = "16050101", type="huc08")
site_loc <- dataRetrieval::readNWISsite("11434500") %>% 
  st_as_sf(coords = c("dec_long_va", "dec_lat_va"), crs=4326)
ggplot() + 
  geom_sf(data=huc_2014, fill="cornflowerblue") + 
  geom_sf(data=huc_2016, fill="salmon") +
  geom_sf(data=site_loc, shape=17, size=3) +
  theme_bw()
```

</details>

image

**Pull request merged**